test/test_import_compatibility.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2024-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test that public API imports are consistent with our public docs."""
+from __future__ import annotations
+
+
+def test_imports():
+    import pymongo
+    from pymongo import (
+        auth_oidc,
+        change_stream,
+        client_options,
+        client_session,
+        collation,
+        collection,
+        command_cursor,
+        cursor,
+        database,
+        driver_info,
+        encryption,
+        encryption_options,
+        errors,
+        event_loggers,
+        mongo_client,
+        monitoring,
+        operations,
+        pool,
+        read_concern,
+        read_preferences,
+        results,
+        server_api,
+        server_description,
+        topology_description,
+        uri_parser,
+        write_concern,
```

Comment:
I think we need to actually import everything from each module:
```
from pymongo.collation import Collation, CollationStrength, ...
...
```

Reply:
Everything public? Or is `from pymongo.collation import *` sufficient?

Reply:
Yeah everything, that way we know every object we expect to be there is actually importable.

Reply:
Wouldn't using `from x import *` be sufficient then? Since that imports everything without a `_` in front?

Reply:
We also have `test_default_exports.py`, perhaps that just needs to be updated?

Reply:
Good point: `test_default_exports` checks that all of our top-level modules exist, but doesn't check that importing them exposes the correct public APIs. Adding this new import test to that file makes sense to me.